app/components/OrganigramInput/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useCallback } from 'react';
+import { _cs } from '@togglecorp/fujs';
+import { PartialForm } from '@togglecorp/toggle-form';
+import OrgTree from 'react-org-tree';
+import {
+    Card,
+} from '@the-deep/deep-ui';
+import styles from './styles.css';
+
+import { OrganigramValue, OrganigramDatum } from '#types/newAnalyticalFramework';
+
+interface Props<N extends string> {
+    className?: string;
+    name: N;
+    value: OrganigramValue | null | undefined;
+    onChange: (value: OrganigramValue | undefined, name: N) => void,
+    options: PartialForm<OrganigramDatum> | null | undefined;
+    readOnly?: boolean;
+    disabled?: boolean;
+}
+
+function OrganigramInput<N extends string>(props: Props<N>) {
+    const {
+        className,
+        name,
+        value,
+        onChange,
+        disabled,
+        readOnly,
+        options,
+    } = props;
+
+    const handleClick = useCallback((_: Event, data: OrganigramDatum) => {
+        if (value?.some((v) => v === data.clientId)) {
```

Comment:
Can there be multiple data.clientId in `value`? Why are we using some instead of find?

Reply:
yes there will be multiple values in `value` props. The value props contains list of `clientId` right now. Maybe later we need to pass idSelector as there will be `id` passed from the server. Isn't `array.some` more semantic to check if item exists in an array and use it inside `if` condition.